test/lib/cql_test_env.cc

```diff
@@ -591,7 +588,9 @@ class single_node_cql_env : public cql_test_env {
 
             sharded<locator::shared_token_metadata> token_metadata;
             locator::token_metadata::config tm_cfg;
-            tm_cfg.topo_cfg.this_host_id = cfg->host_id;
+            // FIXME: should we update this later? #14638
+            // main.cc doesn't update it either, we simulate the behavior.
```

Comment:
It is used only by unit tests.
It's more than ok for cql_test_env to mimic main

Reply:
I don't think this deserves a FIXME.
A comment explaining that cql_test_env is doing the same as main would be enough